### ISSUE TITLE
Add `splice`

### ISSUE THREAD

**Issue by nnethercote** [Gankra/thin-vec#39]
`ThinVec` currently lacks the `splice` method. In https://github.com/rust-lang/rust/pull/104754 I converted a bunch of `Vec` instances in rustc to `ThinVec` for some performance wins. I had to change one use of `splice` like this:
```
-                alternatives.splice(idx..=idx, inner);
+ 
+                // Equivalent to `alternatives.splice(idx..=idx, inner);`.
+                alternatives.remove(idx);
+                for p in inner.into_iter().rev() {
+                    alternatives.insert(idx, p);
+                }
```
which could cause quadratic behaviour.

**Comment by nnethercote** [comment]
There is even a `TODO: implement splice?` comment in the code.

**Comment by Gankra** [comment]
https://github.com/Gankra/thin-vec/pull/41

**Comment by marcospb19** [comment]
Hey :wave:, should this be closed due to #41?
